### PR TITLE
update contributing

### PR DESCRIPTION
**Pull request by lostmygithubaccount**
- setup.py -> setup-workspace.py
- edit max nodes
- run deepspeed example on k80s
- edits
- run readme
- fix deepspeed
- edit readme
- minor edits, fix typo
- remove line
- remove line
- e
- switch deepspeed example back
- update contrib
- saving
- minor edit
- saving

# PR into Azure/azureml-examples

## Checklist

I have:

- [x] read and followed the contributing guidelines
- [x] ran `python readme.py` after making changes (per contributing guidelines)
- [x] added required testing (per contributing guidelines)

## Changes

- CONTRIBUTING.md

fixes #

addresses #253 and various other concerns from many :)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -16,128 +16,158 @@ This project has adopted the [Microsoft Open Source Code of Conduct](https://ope
...
-* `tutorials` is for end to end tutorials using AML
+### Unenforced naming
 
-### Enforced naming
+- `environment_name` = "framework-example|tutorial" e.g. "pytorch-example"
+- `experiment_name` = "logical-words-example|tutorial" e.g. "hello-world-tutorial"
+- `compute_name` = "compute-defined-in-setup-workspace.py" e.g. "gpu-K80-2"
+- `ws = Workspace.from_config()`
+- `dstore = ws.get_default_datastore()`
+- `ds = Dataset.File.from_files(...)`
```
Comment (dkmiller) [inline CONTRIBUTING.md:87]: What if there are multiple datasets?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -16,128 +16,158 @@ This project has adopted the [Microsoft Open Source Code of Conduct](https://ope
...
+- `using-*` for learning ML tooling basics and tracking/scaling in the cloud
+- `work-with-*` for integrations with cloud tooling, e.g. `work-with-databricks`, `work-with-synapse`
+- `deploy-*` for advanced deployment scenarios
+- `automl-with-*` for automated ML tutorials
+
+#### Notebooks
+
+A notebook is an example accomplishing something significant in a Jupyter notebook, often written in Python. To qualify to be a notebook, the example must:
+
+- obviously benefit from being a Jupyter notebook
```
Comment (dkmiller) [inline CONTRIBUTING.md:119]: What makes something _not_ obviously benefit from being a Jupyter notebook?

I.e., do you allow "raw markdown / Python" examples?

File: CONTRIBUTING.md (hunk 1 above)
Comment (lostmygithubaccount) [reply CONTRIBUTING.md:87]: `dss` for a list of datasets, `ds1`, `ds2`, `ds3`, etc. for individual datasets, would be my opinion but this is in the unenforced naming section

same for the other objects

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -16,128 +16,158 @@ This project has adopted the [Microsoft Open Source Code of Conduct](https://ope
...
-* `cleanup` runs daily and cleans up AML resources for the testing workspace
-* `smoke` runs hourly and on every push and PR to `main` and performs sanity checks
+- required prerequisites
+- any one-time setup needed by the user
+- any other setup instructions
+- overview of files in the tutorial
+- relevant links
 
-### Miscellaneous
+Tutorials are often, but not required to be, a series of ordered Jupyter notebooks. All Jupyter notebooks must utilize notebook features (i.e. be interactive, have explanation in markdown cells, etc).
```
Comment (dkmiller) [inline CONTRIBUTING.md:106]: Should probably say something about how you expect meaningful explanations in markdown cells.

**Review by dkmiller — COMMENTED** [review]
Overall looks pretty good, but made a couple comments.

**Pull request merged**